pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -248,6 +251,10 @@ func (service *AlertRuleService) ReplaceRuleGroup(ctx context.Context, orgID int
 		}
 	}
 
+	if err := service.checkGroupLimits(group); err != nil {
```

Comment:
this does not cover Insert\Update\Delete rules method that potentially can add\remove rules from groups.

Reply:
Some of these required changes that didn't seem worth it just for a warning log. Most paths here don't load the entire group - even Update just loads a partial group seemingly. It didn't feel worth it to read in all this extra data just for a log message.

Especially since Grafana-owned code generally uses the Group endpoint, the only users who would hit this are consuming the API directly.

Though I guess we'll have to read all this data eventually to enforce the limit for real, won't we?

Reply:
I tried something different, I agree it didn't warn often enough before.

I instead added the check to the state manager `Warm()` function. That way:
- A warning always gets emitted every startup, even if no write happened
- The warm function already lists all rules, so we can easily see all group sizes without needing to query additional data from the database.
- It runs on startup only, and not on the periodic fetch, so it's not spamming the warning every couple minutes.

I think this is an okay workaround since users of this API will now get at least one message (all users will at startup). It doesn't solve how we'll enforce the limit on the provisioning API, but we can save that for a future change, the only goal for now is to emit a warning but not enforce anything. What do you think?